crypto/x509/x509_lu.c

```diff
@@ -118,6 +118,23 @@ int X509_LOOKUP_by_alias(X509_LOOKUP *ctx, X509_LOOKUP_TYPE type,
     return ctx->method->get_by_alias(ctx, type, str, len, ret);
 }
 
+int X509_LOOKUP_set0_method_data(X509_LOOKUP *ctx, void *data)
+{
+    ctx->method_data = (char*)data;
```

Comment:
I don't think that cast is required. And given the usage pattern it appears appropriate to change method_data declaration to void *... To denote that it's in fact an opaque data to X509_LOOKUP...

Reply:
Yup...  there is a [previous examples of the same](https://github.com/openssl/openssl/blob/master/crypto/x509/by_dir.c#L113), I can shave that off while I''m at it...